backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/RemoveDiskCommand.java

```diff
@@ -728,4 +733,16 @@ private Guid getQuotaId() {
     @Override
     public void addQuotaPermissionSubject(List<PermissionSubject> quotaPermissionList) {
     }
+
+    private DiskStorageType getDiskStorageType() {
+        if (getParameters().getDiskStorageType() == null) {
+            getParameters().setDiskStorageType(getDisk().getDiskStorageType());
```

Comment:
I think you need persistCommandIfNeeded after this, at the end of the executeCommand